src/main/java/org/launchcode/water_garden_tour/controllers/MapController.java

```diff
@@ -14,9 +19,22 @@
     GardenRepository gardenRepository;
 
     @RequestMapping("map")
-    public String map(Model model) {
-        model.addAttribute("gardens", gardenRepository.findAll());
-        model.addAttribute("title", "Garden Map View");
+    public String map(Model model, @RequestParam(required=false, value="gardenId") Integer gardenId) {
```

Comment:
Here I added gardenId as an optional request parameter. There are two types of links going to this controller now, 1 from standard map on the navbar and a second from the individual gardens on cards through the individual view buttons. These buttons send the gardenId as a request parameter, but, importantly, the main navbar link does not have this request param (nor does the "all garden" button now on the map.html template). So by checking if gardenId is null, we can route two different ways in this controller, one for all gardens, and the other for one garden. Ideally these should probably be two sites but I had some trouble with the javascript that way and this was what I got to work while our resident google maps expert is out haha.